client.go

```diff
@@ -102,7 +104,7 @@ func NewClient(config *Config) (client *Client, err error) {
 
 	tr := http.DefaultTransport.(*http.Transport)
 	tr.TLSClientConfig = &tls.Config{InsecureSkipVerify: config.SkipSslValidation}
-	ctx = context.WithValue(ctx, oauth2.HTTPClient, &http.Client{Transport: tr})
+	ctx = context.WithValue(ctx, oauth2.HTTPClient, &http.Client{Transport: tr, Timeout: config.Timeout})
```

Comment:
hmm. From my reading of the source is: if there is a http client in the context it will reuse that one. Otherwise if there's always a new http client set, then the timeout setting wouldn't help anyway, no?